crawler.js

```diff
@@ -410,45 +429,58 @@ export class Crawler {
 
       if (this.params.behaviorOpts) {
         if (!page.isHTMLPage) {
-          this.logger.info("Skipping behaviors for non-HTML page", data, "behavior");
+          this.logger.info("Skipping behaviors for non-HTML page", logDetails, "behavior");
         } else {
-          await Promise.allSettled(
-            page.frames().
-              filter(frame => this.shouldRunBehavior(frame)).
-              map(frame => evaluateWithCLI(frame, "self.__bx_behaviors.run();"))
-          );
-
-          // also wait for general net idle
-          await this.netIdle(page);
+          const behaviorTimeout = this.params.behaviorTimeout / 1000;
+          this.logger.info("Behaviors started", {behaviorTimeout, ...logDetails}, "behavior");
+          const res = await Promise.race([
+            this.sleep(behaviorTimeout),
+            Promise.allSettled(
+              page.frames().
+                filter(frame => this.shouldRunBehavior(frame, logDetails)).
+                map(frame => evaluateWithCLI(frame, "self.__bx_behaviors.run();", logDetails, "behavior"))
+            )
+          ]);
+          if (res && res.length) {
+            this.logger.info("Behaviors finished", {finished: res.length, ...logDetails}, "behavior");
+          } else {
+            this.logger.error("Behaviors timed out", logDetails, "behavior");
+          }
         }
       }
 
-      this.logger.info("Page graph data for successfully crawled page", data, "pageGraph");
-
-      await this.writeStats();
+      this.logger.info("Page finished", logDetails, "pageStatus");
 
       await this.checkLimits();
 
       await this.serializeConfig();
 
     } catch (e) {
-      this.logger.error(`Error crawling page ${data.url}`, e);
+      this.logger.error("Page Errored", {...e, ...logDetails}, "pageStatus");
       await this.markPageFailed(page);
     }
   }
 
-  async shouldRunBehavior(frame) {
+  async shouldRunBehavior(frame, logDetails) {
     if (!frame.parentFrame()) {
       return true;
     }
 
     const url = frame.url();
 
+    let res;
+
     if (url === "about:blank") {
-      return false;
+      res = true;
```

Comment:
I'm wondering about the change in logic here - `about:blank` would result in an empty page in the iframe, no?

Reply:
Is the idea that the iframe may be loaded via JS so we should run the behavior just in case?

Reply:
yes, exactly - was trying this out in case it resulted in additional content, but was inconclusive. maybe only ads do this?

Reply:
Reverted back to false for now, let's see if its needed.